src/python/include/Alpha_complex_factory.h

```diff
@@ -0,0 +1,144 @@
+  using Exact_kernel = CGAL::Epeck_d<CGAL::Dynamic_dimension_tag>;
+  using Point_exact_kernel = typename Exact_kernel::Point_d;
+
+ public:
+  Exact_Alphacomplex_dD(const std::vector<std::vector<double>>& points, bool exact_version)
+      : exact_version_(exact_version) {
+    ac_exact_ptr_ = std::make_unique<Alpha_complex<Exact_kernel>>(
+      boost::adaptors::transform(points, pt_cython_to_cgal<Point_exact_kernel>));
+  }
+
+  std::vector<double> get_point(int vh) {
+    Point_exact_kernel const& point = ac_exact_ptr_->get_point(vh);
+    return pt_cgal_to_cython(point);
+  }
+
+  void create_simplex_tree(Simplex_tree_interface<>* simplex_tree, double max_alpha_square,
+                           bool default_filtration_value){
+    ac_exact_ptr_->create_complex(*simplex_tree, max_alpha_square, exact_version_, default_filtration_value);
+  }
+
+ private:
+  bool exact_version_;
+  std::unique_ptr<Alpha_complex<Exact_kernel>> ac_exact_ptr_;
+};
+
+class Inexact_Alphacomplex_dD : public Abstract_alpha_complex {
+ private:
+  using Inexact_kernel = CGAL::Epick_d<CGAL::Dynamic_dimension_tag>;
+  using Point_inexact_kernel = typename Inexact_kernel::Point_d;
+
+ public:
+  Inexact_Alphacomplex_dD(const std::vector<std::vector<double>>& points, bool exact_version)
+      : exact_version_(exact_version) {
+    ac_inexact_ptr_ = std::make_unique<Alpha_complex<Inexact_kernel>>(
+      boost::adaptors::transform(points, pt_cython_to_cgal<Point_inexact_kernel>));
+  }
+
+  std::vector<double> get_point(int vh) {
+    Point_inexact_kernel const& point = ac_inexact_ptr_->get_point(vh);
+    return pt_cgal_to_cython(point);
+  }
+  void create_simplex_tree(Simplex_tree_interface<>* simplex_tree, double max_alpha_square,
+                           bool default_filtration_value){
+    ac_inexact_ptr_->create_complex(*simplex_tree, max_alpha_square, exact_version_, default_filtration_value);
+  }
+
+ private:
+  bool exact_version_;
+  std::unique_ptr<Alpha_complex<Inexact_kernel>> ac_inexact_ptr_;
+};
+
+template <complexity Complexity>
+class Alphacomplex_3D : public Abstract_alpha_complex {
+ private:
+  using Point_3 = typename Alpha_complex_3d<Complexity, false, false>::Bare_point_3;
+
+  static Point_3 pt_cython_to_cgal_3(std::vector<double> const& vec) {
+    return Point_3(vec[0], vec[1], vec[2]);
+  }
+
+ public:
+  Alphacomplex_3D(const std::vector<std::vector<double>>& points) {
+    alpha3d_ptr_ = std::make_unique<Alpha_complex_3d<Complexity, false, false>>(
+      boost::adaptors::transform(points, pt_cython_to_cgal_3));
+  }
+
+  std::vector<double> get_point(int vh) {
+    Point_3 const& point = alpha3d_ptr_->get_point(vh);
+    return pt_cgal_to_cython(point);
+  }
+
+  void create_simplex_tree(Simplex_tree_interface<>* simplex_tree, double max_alpha_square,
+                           bool default_filtration_value){
+    alpha3d_ptr_->create_complex(*simplex_tree, max_alpha_square);
+    if (default_filtration_value) {
+      // TODO
```

Comment:
Ah, yes, that complicates things. In that case, we don't want to insert the points in the Alpha_complex_3d, we only want the delaunay triangulation. So for instance, in the constructor, we could store the points and only build things in create_complex. Or we could in the constructor build the Alpha_shape_3::Triangulation, and in create_complex build the Alpha_shape_3 from the triangulation. Or we can iterate on the simplices and put all of the filtration values to NaN here (I believe we do guarantee a value of NaN, not just don't care?), but that doesn't seem very efficient.

Reply:
Fixing #73 would help to go with your last proposition (construct the alpha shape and finally reset_filtration to NaN).

Reply:
Yes, but that's the "solution" I like the least...
Did you consider the version where the constructor does nothing (just stores the points for later use), and create_complex does everything? It may also simplify the design, make it unnecessary to have many different classes.